src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4753,22 +4753,32 @@ WorkerLimitCount(Node *limitCount, Node *limitOffset, OrderByLimitReference
 	if (workerLimitNode != NULL && limitOffset != NULL)
 	{
 		Const *workerLimitConst = (Const *) workerLimitNode;
-		Const *workerOffsetConst = (Const *) limitOffset;
-		int64 workerLimitCount = DatumGetInt64(workerLimitConst->constvalue);
-		int64 workerOffsetCount = DatumGetInt64(workerOffsetConst->constvalue);
-
-		workerLimitCount = workerLimitCount + workerOffsetCount;
-		workerLimitNode = (Node *) MakeIntegerConstInt64(workerLimitCount);
+		if (!workerLimitConst->constisnull)
+		{
+			/* Only update the worker limit if the const is not null.*/
+			Const *workerOffsetConst = (Const *) limitOffset;
+			int64 workerLimitCount = DatumGetInt64(workerLimitConst->constvalue);
+			int64 workerOffsetCount = DatumGetInt64(workerOffsetConst->constvalue);
```

Comment:
Please add a comment on how it is ok to use this even when `OFFSET NULL` is used. See earlier comment for details.

Reply:
Done